CMakeLists.txt

```diff
@@ -36,6 +36,26 @@ if(NOT TARGET readerwriterqueue)
     FetchContent_MakeAvailable(ReaderWriterQueue)
 endif()
 
+if (NOT TARGET farbot)
```

Comment:
I'm completely spitballing here. I just want to make sure before we extend to 2 backends that we have a plan for when we inevitably need to add a 3rd

Reply:
I understand your concern. Once thing I would point out is that we would probably like to have those options alongside and not exclusive - one would like to have both kinds of loggers (also to avoid possible build config conflicts between sub-projects).

(BTW I didn't replace readerwriterqueue with concurrentqueue, so only farbot version is added. I can add that too if you want, but it's not going to be RT log anymore ;-) )

Reply:
I'm digging into this just a little bit and I remember one of the sticking points that prevented me from doing this in the past.


If you change the args to have a QueueType:

```
template <typename LogData, size_t MaxNumMessages, size_t MaxMessageLength,
          std::atomic<std::size_t>& SequenceNumber, typename QueueType>
```

This becomes tricky because QueueType needs to depend on `InternalLogData`.

(I'm not sure immediately how to re-arrange to get around this problem)